picoquic/packet.c

```diff
@@ -148,53 +68,45 @@ int picoquic_parse_packet_header(
 {
     int ret = 0;
 
-    /* Initialize the PH structure to zero, but version index to -1 (error) */
-    memset(ph, 0, sizeof(picoquic_packet_header));
-    ph->version_index = -1;
-
-    /* Is this a long header of a short header? -- in any case, we need at least 17 bytes */
-    if ((bytes[0] & 0x80) == 0x80) {
-        if (length < 6) {
+    if (length < 6) {
+        ret = -1;
+    }
+    else {
+        /* The bytes at position 1..4 describe the version */
+        ph->vn = PICOPARSE_32(bytes + 1);
+        if (!picoquic_is_header_version_supported(ph->vn)) {
             ret = -1;
```

Comment:
As I say above, we can just skip this and assume that we only have supported version. Everything else would be a parsing error.

Reply:
Done.